test_runner/src/main/kotlin/ftl/ios/xctest/XcTestData.kt

```diff
@@ -0,0 +1,98 @@
+package ftl.ios.xctest
+
+import com.dd.plist.NSDictionary
+import ftl.args.ArgsHelper.calculateShards
+import ftl.args.IosArgs
+import ftl.ios.xctest.common.XcTestRunVersion
+import ftl.ios.xctest.common.XcTestRunVersion.V1
+import ftl.ios.xctest.common.XcTestRunVersion.V2
+import ftl.ios.xctest.common.XctestrunMethods
+import ftl.ios.xctest.common.getXcTestRunVersion
+import ftl.ios.xctest.common.mapToRegex
+import ftl.ios.xctest.common.parseToNSDictionary
+import ftl.shard.Chunk
+import ftl.shard.testCases
+import ftl.util.FlankTestMethod
+import java.io.File
+
+data class XcTestRunData(
+    val rootDir: String,
+    val nsDict: NSDictionary,
+    val version: XcTestRunVersion = nsDict.getXcTestRunVersion(),
+    val shardTargets: Map<String, List<XctestrunMethods>> = emptyMap(),
+    val shardChunks: Map<String, List<Chunk>> = emptyMap()
+)
+
+fun IosArgs.calculateXcTestRunData(): XcTestRunData {
+    val xcTestRunFile = File(xctestrunFile)
```

Comment:
I think it won't work for files provided via `gs://` path

Reply:
That was not working previously - https://github.com/Flank/flank/blob/074cf0e1bc0afafd4dd99820e60df2fe427cf5a3/test_runner/src/main/kotlin/ftl/ios/Xctestrun.kt#L107

Reply:
This is the source of changes https://github.com/Flank/flank/commit/501ea710b7c949a294f4a35e75ccfa619655e90e

Reply:
We can implement support for `gs` scheme in another PR.